samples/nrf9160/azure_iot_hub/src/main.c

```diff
@@ -127,10 +127,12 @@ static void on_evt_direct_method(struct azure_iot_hub_method *method)
 	printk("Method name: %s\n", method->name);
 	printf("Payload: %.*s\n", method->payload_len, method->payload);
 
-	strncpy(method_data.request_id, method->rid,
-		sizeof(method_data.request_id));
+	strncpy(method_data.request_id, method->rid, sizeof(method_data.request_id));
+	method_data.request_id[sizeof(method_data.request_id) - 1] = '\0';
```

Comment:
I think this could overwrite the last character. Using 
`strncpy(method_data.request_id, method->rid, sizeof(method_data.request_id) - 1)` 
would avoid that

Reply:
I thought of that but with a closer look at `strncpy` the copying does not occur for the last character. If I'm not mistaken?

Reply:
If `method->rid` is longer than the size of `method_data.request_id`, `strncpy` will copy up to `sizeof(method_data.request_id)` characters, so the last one will be overwritten on the line below. There's an explanation in the Notes section here: https://linux.die.net/man/3/strncpy

Reply:
Will fix.